libs/mstable/flatcoin/src/state.ts

```diff
@@ -1,34 +1,115 @@
-import { useCallback, useState } from 'react';
+import { useEffect, useState } from 'react';
 
+import { useSearch } from '@tanstack/react-location';
+import produce from 'immer';
 import { createContainer } from 'react-tracked';
 
+import { PositionType } from './types';
+
 import type { Dispatch, SetStateAction } from 'react';
 
-import type { FlatcoinPageState, FlatcoinTradingPanelType } from './types';
+import type { FlatcoinRoute, Position } from './types';
 
-const initialState: FlatcoinPageState = {
-  type: 'stable',
+type FlatcoinState = {
+  data: {
+    apy?: string;
+    tvl?: string;
+  };
+  configs: Record<Lowercase<PositionType>, any>;
+  positions?: Position[];
+  type: Lowercase<PositionType>;
```

Comment:
Why do we need `Lowercase` here? probably `PositionType` can be used here and in all other places without lowercase to simplify things. Or we can make `PositionType` lowercased by default
(can be done in scope of next pr)